src/Auth0.php

```diff
@@ -317,35 +304,19 @@ public function __construct(array $config)
             $this->dontPersist('id_token');
         }
 
-        $session_base_name = ! empty( $config['session_base_name'] ) ? $config['session_base_name'] : SessionStore::BASE_NAME;
-
-        if (isset($config['store'])) {
-            if ($config['store'] === false) {
-                $emptyStore = new EmptyStore();
-                $this->setStore($emptyStore);
-            } else {
-                $this->setStore($config['store']);
-            }
-        } else {
-            $sessionStore = new SessionStore($session_base_name);
-            $this->setStore($sessionStore);
+        $this->store = $config['store'] ?? null;
+        if ($this->store === false) {
+            $this->store = new EmptyStore();
+        } else if (! $this->store instanceof StoreInterface) {
+            $this->store = new SessionStore();
```

Comment:
Refactor only, no expected functional changes.

Reply:
If `$config['store']` is `null`, is the intention for it to fall into the condition on line 300? I'm assuming that the check on 298 would return `false` in this case. If so, the code isn't very clear here. At first glance I thought that the check on 298 is wrong and it should be `$this->store === null`, but after looking deeper I just have more questions.

Reply:
Thanks for this. I'll take a step back and look at it again. It's more clear than it was before at least 😆